crates/bevy_input/src/input.rs

```diff
@@ -43,29 +43,42 @@ use bevy_ecs::schedule::State;
 ///[`DetectChangesMut::bypass_change_detection`]: bevy_ecs::change_detection::DetectChangesMut::bypass_change_detection
 #[derive(Debug, Clone, Resource, Reflect, FromReflect)]
 #[reflect(Default, FromReflect)]
-pub struct Input<T: Clone + Eq + Hash + Send + Sync + 'static> {
+pub struct Input<T: FromReflect + Clone + Eq + Hash + Send + Sync + 'static> {
     /// A collection of every button that is currently being pressed.
     pressed: HashSet<T>,
     /// A collection of every button that has just been pressed.
     just_pressed: HashSet<T>,
     /// A collection of every button that has just been released.
     just_released: HashSet<T>,
+    /// To map logical keys with their respecting physical keys for reliability.
+    dynamic_map_value: HashMap<T, T>,
 }
 
-impl<T: Clone + Eq + Hash + Send + Sync + 'static> Default for Input<T> {
+impl<T: FromReflect + Clone + Eq + Hash + Send + Sync + 'static> Default for Input<T> {
```

Comment:
To allow `FromReflect` on `Input<T>`, T needs to be `FromReflect` too because of the `HashMap`, it's a bit unfortunate because it adds a constraint on `T`.